cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_auth_method.cc

```diff
@@ -58,9 +58,9 @@ class NoOpClientAuthHandler : public arrow::flight::ClientAuthHandler {
 
   arrow::Status Authenticate(arrow::flight::ClientAuthSender* outgoing,
                              arrow::flight::ClientAuthReader* incoming) override {
-    // Write a blank string. The server should ignore this and just accept any Handshake
+    // Return OK Status. The server should ignore this and just accept any Handshake
     // request.
-    return outgoing->Write(std::string());
+    return arrow::Status::OK();
```

Comment:
After the original flightsql-odbc code has been written, Arrow API changed and now would need `OK` to be returned directly instead of writing strings